npc/re/merchants/eden_market.txt

```diff
@@ -0,0 +1,1843 @@
+		mes "Come by anytime, if you want to go to the Eden Market Group!";
+		close;
+	}
+}
+
+//Eden Market Group Guide
+paramk,101,22,3	script	Market Group Guide#info	4_F_ZONDAGIRL,{
+	mes "[Guide]";
+	mes "Hello. Welcome to Eden Market Group, the official marketplace run by the continental merchant coalition, supported by the Eden Group.";
+	next;
+	switch (select("About when we're open.:About Special Products.:About Eden Market Group.:Cancel.")) {
+	case 1:
+		mes "[Guide]";
+		mes "Eden Market Group is a marketplace that contains many different shops.";
+		next;
+		mes "[Guide]";
+		mes "Wednesday is an official holiday designated by the marketplace; most shops are closed on this day.";
+		next;
+		mes "[Guide]";
+		mes "However, depending on the owner's wishes, and the type of items being offered, some shops remain partially open even on the holiday.";
+		next;
+		mes "[Guide]";
+		mes "And, some shops offer special items on certain days of the week. Keep that in mind.";
+		close;
+	case 2:
+		mes "[Guide]";
+		mes "Some shops will sell different items on weekends, certain days of the week, or even during certain hours; these items are different than their usual daily items.";
+		next;
+		mes "[Guide]";
+		mes "'Wonderful Friends' and 'Strong Friends' will sell one rare item each weekend.";
+		mes "'Miner's Coalition' will sell the anvil for blacksmithing only on weekends.";
+		next;
+		mes "[Guide]";
+		mes "I cannot tell you what items individual shops sell, because they are all different.";
+		mes "Why don't you find out for yourself by visiting often?";
+		close;
+	case 3:
+		mes "[Guide]";
+		mes "There are many people with the soul of a merchant that would like to sell their items to people in need of such.";
+		mes "Eden Market Group has designated a place for such people to buy and sell.";
+		next;
+		mes "[Guide]";
+		mes "We provide the means for goods that would otherwise have been contraband to be bought and sold in a fair and transparent manner, working in conjunction with the merchant coalitions and adventurers.";
+		next;
+		mes "[Guide]";
+		mes "Some people mutter that Eden Market Group is a black market... Haha, what a silly idea...";
+		mes "We are merely a merchant coalition.";
+		next;
+		mes "[Guide]";
+		mes "A portion of our profits go towards running the Eden Group, an adventurer's guild.";
+		mes "You can sell the loot from your adventures here, and the profits we make go towards providing services that benefit you. We are killing two birds with one stone, really.";
+		next;
+		mes "[Guide]";
+		mes "Keep in mind that limited quantity items will never go on sale!";
+		close;
+	case 4:
+		mes "[Guide]";
+		mes "Have a good time!";
+		close;
+	}
+
+OnTimer60000:
+	npctalk "Welcome. This is the Eden Market Group.";
+
+OnInit:
+	initnpctimer;
+	end;
+}
+
+//Guards
+paramk,103,17,3	script	Market Guard#1	4_M_YOUNGKNIGHT,{
+	callfunc("F_CoolDown", 5);
+	switch ((atoi(strnpcinfo(NPC_NAME_HIDDEN)) + 1) / 2) {
```

Comment:
`NPC_NAME_HIDDEN` isn't supported. We should think about adding these consts later. And why would we divide by 2 when we have 6 cases and 6 npcs?

Reply:
Okay. I'll change it

Reply:
I already added a follow up commit. Hope that's okay for you.